### PR TITLE
Fix connected site styling

### PR DESCRIPTION
**Pull request by jusleg**
## Explanation

Updates the connected site UI to fix some styling issues.

* fix connected site icon shrinking horizontally when the site has a long domain
* vertically align connected site origin in the connected site row
* fix disconnect button from changing size on hover

### Before

https://user-images.githubusercontent.com/4406751/182301292-3106be26-0ad4-41ac-930a-ff8333a81281.mov

### After

https://user-images.githubusercontent.com/4406751/182301307-ea10e284-3ad8-428a-b6f6-d7504e9c14cb.mov

## Manual Testing Steps

- open extension
- go to connected sites
- hover over disconnect button

## Pre-Merge Checklist

- [x] PR template is filled out
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [ ] PR is linked to the appropriate GitHub issue
- [ ] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [ ] Manual testing complete & passed
- [ ] "Extension QA Board" label has been applied

### CODE REVIEW

**Review by brad-decker — DISMISSED** [review]
LGTM, thanks for the contribution!

File: ui/pages/connected-sites/index.scss [hunk 1]
```diff
@@ -24,10 +24,12 @@
...
-  a,
-  a:hover {
+  a {
     @include H6;
+  }
 
+  a,
+  a:hover {
     color: var(--color-primary-default);
     cursor: pointer;
```
Comment (georgewrmarshall) [inline ui/pages/connected-sites/index.scss:34]:
```suggestion
  a {
    @include H6;
    cursor: pointer;
    color: var(--color-primary-default);
  }
  
  a:hover {
    color: var(--color-primary-alternative);
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Great update! Could we just add `var(--color-primary-alternative)` as the color for the hover state for the disconnect action?

File: ui/pages/connected-sites/index.scss (hunk 1 above)
Comment (georgewrmarshall) [reply ui/pages/connected-sites/index.scss:34]: Above suggestion will need testing locally 🙏

**Review by adonesky1 — DISMISSED** [review]

File: ui/pages/connected-sites/index.scss (hunk 1 above)
Comment (jusleg) [reply ui/pages/connected-sites/index.scss]: Updated demo: 

https://user-images.githubusercontent.com/4406751/183259333-b18771b4-08c4-4131-a91a-9d0bc2377cae.mov

**Review by brad-decker — DISMISSED** [review]

File: ui/pages/connected-sites/index.scss (hunk 1 above)
Comment (georgewrmarshall) [reply ui/pages/connected-sites/index.scss]: Amazing thanks @jusleg 🙏

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Thanks for the contribution 💯 🙏

File: ui/components/ui/site-icon/site-icon.js [hunk 2]
```diff
@@ -3,16 +3,20 @@ import PropTypes from 'prop-types';
...
       <IconWithFallback icon={icon} name={name} size={iconSize} />
     </IconBorder>
   );
 }
 
 SiteIcon.propTypes = {
+  /**
+   * Additional className to add to the root element of SiteIcon
+  */
+  className: PropTypes.string,
```
Comment (brad-decker) [inline ui/components/ui/site-icon/site-icon.js:19]:
```suggestion
  /**
   * Additional className to add to the root element of SiteIcon
   */
```

For alignment lint failure

Comment (jusleg) [reply ui/components/ui/site-icon/site-icon.js]: Thanks for pushing a fix. The suggested change removed the `SiteIcon.propTypes = {`and caused the build to fail. I added it back to your commit and it should now pass the linting step.

**Comment by brad-decker** [comment]
I'm fighting CI to run tests on this branch. Once i can get it all green i'll get it merged @jusleg

File: ui/pages/connected-sites/index.scss [hunk 3]
```diff
@@ -24,11 +24,13 @@
     margin-top: 15px;
   }
 
-  a,
-  a:hover {
+  a {
     @include H6;
-
```
Comment (brad-decker) [inline ui/pages/connected-sites/index.scss:30]:
```suggestion

```

File: ui/pages/connected-sites/index.scss [hunk 4]
```diff
@@ -24,11 +24,13 @@
...
     color: var(--color-primary-default);
     cursor: pointer;
   }
+
```
Comment (brad-decker) [inline ui/pages/connected-sites/index.scss:32]:
```suggestion
  a {
    @include H6;

    color: var(--color-primary-default);
    cursor: pointer;
  }
```

File: ui/pages/connected-sites/index.scss [hunk 5]
```diff
@@ -24,11 +24,13 @@
...
+  a:hover {
+    color: var(--color-primary-alternative);
+  }
```
Comment (brad-decker) [inline ui/pages/connected-sites/index.scss:35]:
```suggestion
  a {
    @include H6;
    
    color: var(--color-primary-default);
    cursor: pointer;
  }

  a:hover {
    color: var(--color-primary-alternative);
  }
```

**Review by brad-decker — APPROVED** [review]

**Review by georgewrmarshall — APPROVED** [review]
Nice! Thanks for the addition there @brad-decker

**Pull request merged**